zcash_primitives/src/zip32/fingerprint.rs

```diff
@@ -0,0 +1,81 @@
+//! Seed Fingerprints according to ZIP 32
+//!
+//! Implements section `Seed Fingerprints` of Shielded Hierarchical Deterministic Wallets (ZIP 32)
+//!
+//! [Section Seed Fingerprints]: https://zips.z.cash/zip-0032#seed-fingerprints
+use blake2b_simd::Params as Blake2bParams;
+
+pub const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
+pub struct SeedFingerprint([u8; 32]);
+
+impl SeedFingerprint {
+    /// Return the seed fingerprint of the wallet as defined in
+    /// <https://zips.z.cash/zip-0032#seed-fingerprints> or None
+    /// if the length of `seed_bytes` is less than 32 or
+    /// greater than 252.
+    pub fn from_seed(seed_bytes: &[u8]) -> Option<SeedFingerprint> {
+        let seed_len = seed_bytes.len();
+
+        if !(32..=252).contains(&seed_len) {
+            return None;
+        }
+
+        let seed_len_byte = if let Ok(len) = u8::try_from(seed_len).map(|l| l) {
+            len
+        } else {
+            return None;
+        };
+
+        match Blake2bParams::new()
+            .hash_length(32)
+            .personal(ZIP32_SEED_FP_PERSONALIZATION)
+            .to_state()
+            .update(&[seed_len_byte])
+            .update(seed_bytes)
+            .finalize()
+            .as_bytes()
+            .try_into()
+        {
+            Ok(blake2b_hash) => Some(SeedFingerprint(blake2b_hash)),
+            Err(_) => None,
+        }
```

Comment:
In general:
* Only use error/`None` returns for cases that can happen as a result of invalid input.
* Use panics for cases that cannot occur or would indicate a programming error. These panics don't have to be documented.
* It can also be okay to use a panic for invalid input (but in this case I agree returning `None` is better). Such cases must be documented in a `Panics` section.
* When unwrapping an `Option` where a panic is wanted in the error case: it's okay to use `.unwrap()` when the reason why the error can't happen is clear from immediate context, otherwise use `.expect(...)`.

```suggestion
        if (32..=252).contains(&seed_len) {
            let seed_len: u8 = seed_len.try_into().unwrap();
            Some(SeedFingerprint(
                Blake2bParams::new()
                    .hash_length(32)
                    .personal(ZIP32_SEED_FP_PERSONALIZATION)
                    .to_state()
                    .update(&[seed_len])
                    .update(seed_bytes)
                    .finalize()
                    .as_bytes()
                    .try_into()
                    .expect("hash length should be 32 bytes"),
            ))
        } else {
            None
        }
```

Reply:
Thank you for the feedback! I added it to the PR and also included a test for when `None` is returned.